pkg/machines/libvirt.es6

```diff
@@ -99,59 +99,82 @@ export default {
-                dispatch(delayPolling(getAllVms()));
+    GET_ALL_VMS ({ connectionName }) {
+        logDebug(`${this.name}.GET_ALL_VMS(connectionName='${connectionName}'):`);
+        if (connectionName) {
+            return dispatch => doGetAllVms(dispatch, connectionName);
+        }
+
+        return dispatch => { // for all connections
+            return cockpit.user().done( loggedUser => {
+                const promises = Object.getOwnPropertyNames(VMS_CONFIG.Virsh.connections)
+                    .filter(
+                        // The 'root' user does not have its own qemu:///session just qemu:///system
+                        // https://bugzilla.redhat.com/show_bug.cgi?id=1045069
+                        connectionName => canLoggedUserConnectSession(connectionName, loggedUser))
+                    .map( connectionName => dispatch(getAllVms(connectionName)));
+
+                return cockpit.all(promises)
+                    .then(() => { // keep polling AFTER all VM details have been read (avoid overlap)
+                        dispatch(delayPolling(getAllVms()));
+                    });
             });
         };
     },
 
-    SHUTDOWN_VM ({ name }) {
+    SHUTDOWN_VM ({ name, connectionName }) {
         logDebug(`${this.name}.SHUTDOWN_VM(${name}):`);
-        return spawnVirsh('SHUTDOWN_VM', 'shutdown', name);
+        return spawnVirsh(connectionName, 'SHUTDOWN_VM', 'shutdown', name);
     },
 
-    FORCEOFF_VM ({ name }) {
+    FORCEOFF_VM ({ name, connectionName }) {
         logDebug(`${this.name}.FORCEOFF_VM(${name}):`);
-        return spawnVirsh('FORCEOFF_VM', 'destroy', name);
+        return spawnVirsh(connectionName, 'FORCEOFF_VM', 'destroy', name);
     },
 
-    REBOOT_VM ({ name }) {
+    REBOOT_VM ({ name, connectionName }) {
         logDebug(`${this.name}.REBOOT_VM(${name}):`);
-        return spawnVirsh('REBOOT_VM', 'reboot', name);
+        return spawnVirsh(connectionName, 'REBOOT_VM', 'reboot', name);
     },
 
-    FORCEREBOOT_VM ({ name }) {
+    FORCEREBOOT_VM ({ name, connectionName }) {
         logDebug(`${this.name}.FORCEREBOOT_VM(${name}):`);
-        return spawnVirsh('FORCEREBOOT_VM', 'reset', name);
+        return spawnVirsh(connectionName, 'FORCEREBOOT_VM', 'reset', name);
     },
 
-    START_VM ({ name }) {
+    START_VM ({ name, connectionName }) {
         logDebug(`${this.name}.START_VM(${name}):`);
-        return spawnVirsh('START_VM', 'start', name);
+        return spawnVirsh(connectionName, 'START_VM', 'start', name);
     }
 };
 
+function canLoggedUserConnectSession (connectionName, loggedUser) {
+    return connectionName !== 'session' || loggedUser.name !== 'root';
+}
+
+function doGetAllVms (dispatch, connectionName) {
+    const connection = VMS_CONFIG.Virsh.connections[connectionName];
+
+    return spawnScript({
+        script: `virsh ${connection.params.join(' ')} -r list --all | awk '$1 == "-" || $1+0 > 0 { print $2 }'`
```

Comment:
Only values from config.es6 can appear here, so I don't think escaping is needed.